src/twisted/internet/test/test_endpoints.py

```diff
@@ -1687,6 +1688,381 @@ class WithResolver(proxyForInterface(
+
+        receiver = self.resolver.resolveHostName(self.receiver, "example.com")
+
+        self.assertIs(receiver, self.receiver)
+
+        self.fakeResolverReturns.errback(Exception())
+
+        self.assertEqual(1, len(logs))
+        self.assertEqual(1, len(self.flushLoggedErrors(Exception)))
+
+        [event] = logs
+        self.assertTrue(event.get("isError"))
+        self.assertTrue(event.get("name", "example.com"))
+        self.assertTrue(event.get("callable", repr(self.fakeResolver)))
+
+        self.assertEqual(1, len(self.resolutionBeganCalls))
+        self.assertEqual(self.resolutionBeganCalls[0].name, "example.com")
+        self.assertFalse(self.addressResolvedCalls)
+        self.assertEqual(1, self.resolutionCompleteCallCount)
+
+
+    def test_resolveNameDelivers(self):
+        """
+        The resolution receiver begins, and resolved hostnames are
+        delivered before it completes.
+        """
+        port = 80
+        ipv4Host = '1.2.3.4'
+        ipv6Host = '1::2::3::4'
+
+        receiver = self.resolver.resolveHostName(self.receiver, "example.com")
+
+        self.assertIs(receiver, self.receiver)
+
+        self.fakeResolverReturns.callback([
+            (AF_INET, SOCK_STREAM, IPPROTO_TCP, '', (ipv4Host, port)),
+            (AF_INET6, SOCK_STREAM, IPPROTO_TCP, '', (ipv6Host, port)),
+        ])
+
+        self.assertEqual(1, len(self.resolutionBeganCalls))
+        self.assertEqual(self.resolutionBeganCalls[0].name, "example.com")
+        self.assertEqual(self.addressResolvedCalls, [
+            IPv4Address("TCP", ipv4Host, port),
+            IPv6Address("TCP", ipv6Host, port)
+        ])
+        self.assertEqual(self.resolutionCompleteCallCount, 1)
+
+
+
+class _HostnameEndpointMemoryReactorMixin(ClientEndpointTestCaseMixin):
+    """
+    Common methods for testing L{HostnameEndpoint} against
+    L{MemoryReactor} instances that do not provide
+    L{IReactorPluggableNameResolver}.
+    """
+
+    def synchronousDeferredToThread(self, f, *args, **kwargs):
+        """
+        A synchronous version of L{deferToThread}.
+
+        @param f: The callable to invoke.
+        @type f: L{callable}
+
+        @param args: Positional arguments to the callable.
+
+        @param kwargs: Keyword arguments to the callable.
+
+        @return: A L{Deferred} that fires with the result of applying
+            C{f} to C{args} and C{kwargs} or the exception raised.
+        """
+        try:
+            result = f(*args, **kwargs)
+        except:
+            return defer.fail()
+        else:
+            return defer.succeed(result)
+
+
+    def expectedClients(self, reactor):
+        """
+        Extract expected clients from the reactor.
+
+        @param reactor: The L{MemoryReactor} under test.
+
+        @return: List of calls to L{IReactorTCP.connectTCP}
+        """
+        return reactor.tcpClients
+
+
+    def connectArgs(self):
+        """
+
```

Comment:
That seems like a bug in twistedchecker. Or an indication there should be some prose here.